src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/FeedbackLogging.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Internal.VisualStudio.Shell.Embeddable.Feedback;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Shell;
+using Xunit.Harness;
+
+namespace Microsoft.VisualStudio.Razor.IntegrationTests
+{
+    public static class VisualStudioLogging
+    {
+        private static bool s_customLoggersAdded = false;
+
+        public const string RazorOutputLogId = "RazorOutputLog";
+        public const string LogHubLogId = "RazorLogHub";
+        public const string ServiceHubLogId = "ServiceHubLog";
+        public const string ComponentModelCacheId = "ComponentModelCache";
+
+        public static void AddCustomLoggers()
+        {
+            // Add custom logs on failure if they haven't already been.
+            if (!s_customLoggersAdded)
```

Comment:
Not sure if having extra loggers under the same name would break things but given this can be called by any number of threads this might be valuable to have in a lock

Reply:
I don't think it's possible (or at least advisable) to run these integration tests in parallel. That said, there's no harm in hardening this a bit and it should be easy.